### DISCUSSION TITLE
How to implement high availability?

### DISCUSSION THREAD

**Discussion by YYXXYYZZ** [questdb/questdb#1802] [Ideas]
hi, folks,
I found no description about high availability from official document, so, how to achieve high availability? 
thanks

**Comment by pswu11** [comment]
Hi @YYXXYYZZ thanks for starting the discussion! Regarding HA, we currently have replication on our roadmap, most likely, in the first half of 2022. What are other features you expect QuestDB to have for high availability? And could you share more about your use case?

**Reply by jgarth** [reply]
@pswu11 

I'm not OP, but maybe it's helpful to share my use case anyway:

I'm running a Kubernetes cluster
I'm using Telegraf to ingest Kubernetes metrics and a tail of all container logs (running as a DaemonSet) and send them to QuestDB
I'm sending product analytics data (user events from a web app) to QuestDB
I'm querying logs from QuestDB when a service causes trouble
I'm querying the product analytics data from QuestDB to graph / show things like "Most popular X in the last 30 days", "DAU", …

As for HA:
I can currently only have one pod running QuestDB.
What I'd like to do is have multiple pods running QuestDB to enable zero-downtime rolling restarts and shield myself from node failure.

As far as expectations go, first of all I'm really glad to see that the QuestDB team is actively working on this – thanks! Also, I have no idea how you plan to implement replication internally, so the following might not be compatible/great with the solution you are working on.

Ideally the high-availability features of QuestDB would have all the comfort features needed to isolate myself from managing replication/HA by hand on the client side, e.g. multi-master capability, or automatic master node election + announcing master readiness. That way my services (Telegraf, product analytics) are agnostic to node/pod failures or upgrades without the need for me to do anything manually when such failures occur, fail-over could all be automated through K8s, if that's even necessary.

**Reply by marregui** [reply]
hi @jgarth thank you for sharing this. Would you be able to share the schema for the logs table in QuestDB? we are solving a similar requirement. Thank you again!

**Reply by CaryChy** [reply]
Hi guys, so far any updates on the HA solution?

**Reply by tris0laris** [reply]
hey, we gave a good round of updates last month on our community meetup: https://www.youtube.com/watch?v=PfjFT78jlfQ&t=13s
You can also track our roadmap here: https://github.com/orgs/questdb/projects/1/views/5
